### DISCUSSION TITLE
Building birt on different platforms

### DISCUSSION THREAD

**Discussion by colinsutton** [eclipse-birt/birt#809] [Q&A] [answered]
I have successfully built BIRT on Windows and on Macos.cocoa.x86_64 on one of my intel macs.
I notice that the macos all-in-one zip file in the latest builds "Artifacts of birt-master #159" is damaged, won't extract (I downloaded a couple of times in case there was a transient error, but it still won't create an eclipse app). 

When I build BIRT on the m1 using eclipse for the m1 it has the error  in manifest line 6 with "org.eclipse.birt.data.oda.pojo.ui cannot be resolved". Is this caused by use of JRE17 (this is the JDK version that supports aarch64). If i delete line 6 it builds everything OK for the intel platforms only.

Is there a plan to add support for macos.cocoa.aarch64? I built the eclipse dev environment which supports m1 macs, but BIRT does not yet know about AARCH64. Maybe for the release after next?

**Comment by wimjongman** [comment]
what is wrong with bundle org.eclipse.birt.data.oda.pojo.ui?

![image](https://user-images.githubusercontent.com/180969/152243026-affd22b6-3434-4cd0-a336-bfb8f7038656.png)

**Reply by colinsutton** [reply]
<img width="1288" alt="screenshot_581" src="https://user-images.githubusercontent.com/18058891/152264683-177d8868-6ef8-42e1-a95c-b156b9f3a25f.png">
<img width="868" alt="screenshot_583" src="https://user-images.githubusercontent.com/18058891/152265230-c24df898-610d-4f55-ac46-afbdd7f32ba4.png">

**Comment by wimjongman** [comment]
You are not showing the manifest for oda.pjo.ui but for oda.pjo.ui.nl1

**Reply by colinsutton** [reply]
I selected the birt.pojo.ui. The pom differs from the one on my intel mac, it shows a different parent:
<img width="830" alt="screenshot_585" src="https://user-images.githubusercontent.com/18058891/152398596-983ce3f9-ac82-4b4d-8b14-a70138d363ae.png">
The bottom of the package explorer tab is different too, BIRT NL and BIRT Parents have extra '>' 
<img width="356" alt="screenshot_586" src="https://user-images.githubusercontent.com/18058891/152400491-aa750474-2ea8-49f7-964b-e95a658d3d69.png">
The macos.cocoa.AARCH64 environment  needs to be added in birt:
<img width="662" alt="screenshot_587" src="https://user-images.githubusercontent.com/18058891/152402772-3c23b8a0-6311-45ae-892f-447780cce66b.png">
And some of the plug-ins. I don't have any knowledge in this area....

**Reply by wimjongman** [reply]
> And some of the plug-ins. I don't have any knowledge in this area....

It is new to everybody.

**Comment by wimjongman** [comment, marked as answer]
Please try an aarch64 download here: https://download.eclipse.org/birt/downloads/drops/snapshot/

**Reply by colinsutton** [reply]
I downloaded the all-in-one builds, but neither run on the relevant platform, both show: ‘“eclipse is damaged and can’t be opened”.
I updated my clone of birt-master and rebuilt on each platform, with the same result.

Note that the macos intel all-in-one in birt-master https://github.com/eclipse/birt/pull/159” would not run either.

**Reply by wimjongman** [reply]
Colin, this is a Mac quarantine issue. Please perform something similar as described here:

https://remainsoftware.com/docs/miworkplace/help/index.jsp#topic/miworkplace.help/manuals/html/MiWorkplace/Installation/InstallationInstallation/InstallationInstallation.html

**Reply by colinsutton** [reply]
Thanks,
That lets the app run. Now testing.

**Reply by colinsutton** [reply]
I have run a few complex reports on windows 10 that were developed using BIRT 4.5 to 4.8  without issues, using the latest runtime and the all-in-one.
I also developed a simple report accessing data from a web service on windows 10, on Macos Big Sur on intel and on MacOS 12.2 Monterey on an M1 mac, didn't see any issues with the report designer.
All the above using the all in one Version: 2022-03 (4.23) Build id: I20220205-1800 for the relevant platform.

As a user, all that is needed is the app icon to be set (I know that's on the way) and the Mac version not to be quarantined.

**Comment by wimjongman** [comment]
Thanks for the tests, Colin!! That is highly appreciated.

**Comment by wimjongman** [comment]
The signing issue is here: https://github.com/eclipse/birt/issues/817
